execution_engine/src/core/runtime/auction_internal.rs

```diff
@@ -98,12 +98,15 @@ where
         account_hash: AccountHash,
         unbonding_purses: Vec<UnbondingPurse>,
     ) -> Result<(), Error> {
-        self.context
-            .metered_write_gs_unsafe(
-                Key::Unbond(account_hash),
-                StoredValue::Unbonding(unbonding_purses),
-            )
-            .map_err(|exec_error| <Option<Error>>::from(exec_error).unwrap_or(Error::Storage))
+        let unbond_key = Key::Unbond(account_hash);
+        if unbonding_purses.is_empty() {
+            self.context.delete_gs_unsafe(unbond_key);
```

Comment:
Deleting a non-existing key is a noop, not a crash, right?

I noticed the `Use with caution` comment, which indicates that at some point someone will use it without caution :)

Reply:
Actually deleting nonexisting is not a noop, it's an error. We first need to read the StoredValue under a Key, then we'll apply a transform on it which could result a Some(new_stored_value) (write, add, identity, etc) or None (delete). Then persisting stored values would either write new data to global state (`Some`) or delete it (`None`)

> Unsafe
It's a convention we have, for storage APIs that are called directly from system contracts. It's not unsafe as in rust's unsafe keyword, it's just the fact that there's no validation, and we don't want to expose this API to users yet, so most if not all the methods I introduced are `pub(crate)` at most.